File: MonsterBall.cpp
//
//  MonsterBall.cpp
//  
//
//  Created by Muhammad Sohibul Maromi on 3/15/13.
//
//

#include "MonsterBall.h"

MonsterBall::MonsterBall(int x, string s) {
    
}

MonsterBall::MonsterBall(const MonsterBall& mb) {
    
}

MonsterBall::~MonsterBall() {
    
}

MonsterBall& MonsterBall::operator=(const MonsterBall& mb) {
    
}

void MonsterBall::execute(Player& p, Monster& m) {
    if(m.isSekarat())
        p.addMonster(m);
    else
        cout << "Tidak bisa ditangkap" << endl;
}

File: Android/model/MonsterBall.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package model;

import pokeranch.Player;

/**
 *
 * @author msmaromi
 */
public class MonsterBall extends Item {
  public MonsterBall(int harga, String nama) {
    super(harga, nama);
  }
  
  public MonsterBall(MonsterBall mb) {
    super(mb);
  }
  
  public void execute(Player p, Monster m) {    
    /*if(m.isSekarat())
      p.addMonster(m);
    else
      System.out.println("monster tidak bisa ditangkap");
      */
  }
}
